src/main/java/axiomq/com/starwars/services/implementations/CharacterServiceImpl.java

```diff
@@ -26,22 +27,30 @@
 
     private final CharacterRepository characterRepository;
 
-
-
     @Override
     public void populateDatabase() {
         String url = "https://swapi.dev/api/people";
-        CharacterExt response = restTemplate.getForObject(url, CharacterExt.class);
         Set<Character> charactersDb = new HashSet<>();
-        List<CharacterInit> characters = new ArrayList<>(response.getResults());
-        characters.forEach(characterInit -> charactersDb.add(characterConverter.toCharacter(characterInit)));
+        while(url!=null) {
+            CharacterExt response = restTemplate.getForObject(url, CharacterExt.class);
+            List<CharacterInit> characters = new ArrayList<>(response.getResults());
```

Comment:
you are not using this list anywhere else after .forEach method, maybe it's better to use response.getResults().forEach() ?